sdks/python/src/stepflow_py/langchain_registry.py

```diff
@@ -0,0 +1,234 @@
+                langchain_input = component_input.input
+                
+                # Create runnable config from StepFlow input
+                stepflow_input_dict = {
+                    "input": component_input.input,
+                    "config": component_input.config or {}
+                }
+                runnable_config = create_runnable_config(stepflow_input_dict, context)
+                
+                # Execute the runnable
+                return await execute_runnable(
+                    runnable,
+                    langchain_input,
+                    config=runnable_config,
+                    execution_mode=exec_mode
+                )
+            
+            # Set up the component input/output types and description
+            langchain_component_executor.__annotations__ = {
+                'component_input': LangChainComponentInput,
+                'context': StepflowContext,
+                'return': Any
+            }
+            
+            # Use provided description or function docstring
+            component_description = description or (f.__doc__.strip() if f.__doc__ else None)
+            if component_description:
+                langchain_component_executor.__doc__ = component_description
+            
+            # Register the component using the server's component method
+            self.component(
+                langchain_component_executor,
+                name=component_name,
+                description=component_description
+            )
+            
+            # Store additional metadata about the LangChain component
+            if not hasattr(self, '_langchain_components'):
+                self._langchain_components = {}
+            
+            self._langchain_components[component_name] = {
+                'factory_function': f,
+                'runnable': runnable,
+                'input_schema': input_schema,
+                'output_schema': output_schema,
+                'execution_mode': execution_mode,
+                'description': component_description
+            }
+            
+            @wraps(f)
+            def wrapper(*args, **kwargs):
+                return f(*args, **kwargs)
+            
+            return wrapper
+        
+        if func is None:
+            return decorator
+        return decorator(func)
+    
+    # Add the method to the server class
+    server_class.langchain_component = langchain_component
+
+
+def get_langchain_components(server) -> Dict[str, Dict[str, Any]]:
+    """
+    Get information about all registered LangChain components.
+    
+    Args:
+        server: The StepFlow server instance
+        
+    Returns:
+        Dictionary mapping component names to their metadata
+    """
+    if not hasattr(server, '_langchain_components'):
+        return {}
+    
+    return server._langchain_components.copy()
+
+
+def list_langchain_components(server) -> list[str]:
```

Comment:
Do we need like an independant add/remove or is it just built at startup time?

Reply:
The decorated functions are registered, so we don't need anything to add/remove.